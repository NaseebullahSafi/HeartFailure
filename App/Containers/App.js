import DebugConfig from '../Config/DebugConfig'
import React, { Component } from 'react'
import { Provider } from 'react-redux'
import RootContainer from './RootContainer'
import createStore from '../Redux'
import {MenuProvider} from "react-native-popup-menu";

// create our store
const store = createStore();

/**
 * Provides an entry point into our application.  Both index.ios.js and index.android.js
 * call this component first.
 *
 * We create our Redux store here, put it into a provider and then bring in our
 * RootContainer.
 *
 * We separate like this to play nice with React Native's hot reloading.
 */
class App extends Component {

  constructor(props) {
    super(props);
  }

  render () {

    return (
      <MenuProvider>
        <Provider store={store}>
          <RootContainer />
        </Provider>
      </MenuProvider>
    );
  }
}

// allow reactotron overlay for fast design in dev mode
export default DebugConfig.useReactotron
  ? console.tron.overlay(App)
  : App
