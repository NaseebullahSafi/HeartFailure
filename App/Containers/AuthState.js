import React, { Component } from 'react'

import firebase from 'react-native-firebase';
import LaunchScreen from "./LaunchScreen";
import OpeningLogoLoader from "./OpeningLogoLoader";

export default class AuthState extends Component {

  constructor() {
    super();

    this.state = {
      loading: true
    }
  }

  /*
* Listen for any state changes and update the component state
* */
  componentDidMount() {
    this.authSubscription = firebase.auth().onAuthStateChanged( user => this.setState({ loading: false, user }) );
  }

  componentWillUnmount() {
    this.authSubscription();
  }

  render () {
    // If the user has not authenticated
    if (this.state.loading)
      return null;

    if (this.state.user)
      return <LaunchScreen/>;

    return <OpeningLogoLoader/>;

  }

}
