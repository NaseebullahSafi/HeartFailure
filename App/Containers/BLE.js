import React, { Component } from 'react';

// predefined component from react
import {
  NativeEventEmitter, NativeAppEventEmitter,
  NativeModules, Platform, AppState, Text,
  PermissionsAndroid, TouchableOpacity
} from 'react-native';


// to convert char to byte array
import { stringToBytes } from 'convert-string';

// the library for BLE connection
import BleManager from "react-native-ble-manager";


// some constant variables
const

  // We need the BleManager from the native module for BLE connection
  BleManagerModule = NativeModules.BleManager,

  // We need emitters to attach them to the app
  bleManagerEmitter = new NativeEventEmitter(BleManagerModule),

  // the mac id of the OpenBCI board
  RFDuinoID = "CF:92:02:BC:0D:CE"
;

/**
 * This component will pair with the board via BLE and try get
 * data from BLE board
 * ==============================================================
 */
export default class BLE extends Component {
  constructor(props) {
    super(props);

    /**
     * The component needs to know whether we are scanning or not. Keep a record
     * of visible devices, and the current app state.
     * @type {{scanning: boolean, peripherals: Map, appState: string}}
     */
    this.state = {
      scanning:false,
      peripherals: new Map(),
      appState: ''
    };


    // deals with devices that are discovered
    this.handleDiscoverPeripheral = this.handleDiscoverPeripheral.bind(this);

    // deals with scanning. I.e. initiate scanning or stop it
    this.handleStopScan = this.handleStopScan.bind(this);

    // deals with the characteristics from BLE
    BLE.handleUpdateValueForCharacteristic = BLE.handleUpdateValueForCharacteristic.bind(this);

    // deals with disconnecting certain device
    this.handleDisconnectedPeripheral = this.handleDisconnectedPeripheral.bind(this);

    // handles the app state
    this.handleAppStateChange = this.handleAppStateChange.bind(this);
  }

  componentDidMount() {

    // first we need to ensure that the scanning happens only after the app is in foreground state
    AppState.addEventListener('change', this.handleAppStateChange);

    // launch the package and ensure that it has been launched and ready
    BleManager.start({showAlert: false}).catch(e => console.error("module initialised"));

    // when a device has been discovered, call the method
    this.handlerDiscover = bleManagerEmitter.addListener (
      'BleManagerDiscoverPeripheral',
      this.handleDiscoverPeripheral
    );

    // when the scan has stopped, call the method
    this.handlerStop = bleManagerEmitter.addListener (
      'BleManagerStopScan',
      this.handleStopScan
    );

    // when a device has been disconnected, call the method
    this.handlerDisconnect = bleManagerEmitter.addListener (
      'BleManagerDisconnectPeripheral',
      this.handleDisconnectedPeripheral
    );

    // when the device emits BLE characteristics, call the method
    this.handlerUpdate = bleManagerEmitter.addListener (
      'BleManagerDidUpdateValueForCharacteristic',
      this.handleUpdateValueForCharacteristic
    );

    // in android, we have to specify to allow BLE
    this.platformPermissionRequest();

    // initiate the scanning for devices
    this.startScan();

    /**
     * Before using BLE, we first have to scan, and then call .connect();
     * and pass the mac id of our board. Once connected, the second phase of BLE
     * is to get our boards services. If services are returned, the next phase will
     * be to get the characteristics. But before doing that, we need to tell the device
     * to notify of any new data.
     */
    BleManager.connect(RFDuinoID).then(() => {
      return BleManager.retrieveServices(RFDuinoID)
    }).then(() => {
      return BleManager.startNotification(
        RFDuinoID,
        '00002220-0000-1000-8000-00805F9B34FB',
        '00002221-0000-1000-8000-00805F9B34FB'
      );
    }).then(() => {
      return NativeAppEventEmitter.addListener(
        'BleManagerDidUpdateValueForCharacteristic', data => {}
        )
    }).catch(e => console.log(e));
  }

  /**
   * In android platform, we have to notify our device to allow the
   * app and search for BLE devices for connection
   * ==============================================================
   */
  platformPermissionRequest = () => {

    // we are in android platform with v.23 and above
    if (Platform.OS === 'android' && Platform.Version >= 23) {

      // to give permission, we need PermissionsAndroid package
      const { PERMISSIONS } = PermissionsAndroid;

      // we first have to modify our android program to allow ACCESS_COARSE_LOCATION
      PermissionsAndroid.check(PERMISSIONS.ACCESS_COARSE_LOCATION).then(result => !result
        ? PermissionsAndroid.requestPermission(
          PERMISSIONS.ACCESS_COARSE_LOCATION
        ).then((result) => {}): null
      );
    }
  };

  /**
   * When the device is connected, we have to tell the app that is
   * active or inactive so that no other devices can join
   * ==============================================================
   * @param nextAppState
   */
  handleAppStateChange = nextAppState => {
    if (this.state.appState.match(/inactive|background/) && nextAppState === 'active') {
      BleManager.getConnectedPeripherals([]).then((peripheralsArray) => {});
    }
    this.setState({appState: nextAppState});
  };

  /**
   * When this component is unmounted, it is crucial to remove all
   * the listeners as it will affect the app performance
   * ==============================================================
   */
  componentWillUnmount() {
    this.handlerDiscover.remove();
    this.handlerStop.remove();
    this.handlerDisconnect.remove();
    this.handlerUpdate.remove();
  }

  /**
   * To disconnect from a particular device, call this method. It
   * will disconnect and update the state object so that the entire
   * component knows that the device is no longer connected
   * ==============================================================
   * @param data
   */
  handleDisconnectedPeripheral = data => {

    // get the list of devices
    let peripherals = this.state.peripherals;

    // get the particular device
    let peripheral = peripherals.get(data.peripheral);

    // if it exists
    if (peripheral) {

      // disconnect it
      peripheral.connected = false;

      // remove it
      peripherals.set(RFDuinoID, peripheral);

      // and push the modified object back to state
      this.setState({peripherals});
    }
  };

  /**
   * This method will print out the characters that the device is
   * trying to emit
   * ==============================================================
   * @param data
   */
  static handleUpdateValueForCharacteristic = data => ({
    peripheral      : data['peripheral'],
    characteristic  : data['characteristic'],
    value           : data['value']
  });

  /**
   * This method stop scanning
   * ==============================================================
   */
  handleStopScan = () => this.setState({ scanning: false });

  /**
   * This method will start scanning
   * ==============================================================
   */
  startScan = () => {

    // first let us check that we arne't already scanning
    if (!this.state.scanning) {

      // let us clear the device list
      this.setState({peripherals: new Map()});

      // initiate the scan and only return max of 3 discovered devices
      BleManager.scan([], 3, true).then(() => this.setState({
        scanning: true
      }));
    }
  };

  /**
   * Return the devices that are connected
   * ==============================================================
   */
  retrieveConnected = () => BleManager.getConnectedPeripherals([]).then((results) => {

    // fetch the device list
    let {peripherals} = this.state;

    for (let i = 0; i < results.length; i++) {
      // get the device
      let peripheral = results[i];

      // set the connection to true
      peripheral.connected = true;

      // put it into list of devices
      peripherals.set(RFDuinoID, peripheral);

      // push that to the state
      this.setState({ peripherals });
    }
  });

  /**
   * Handles devices that are connected
   * ==============================================================
   * @param peripheral
   */
  handleDiscoverPeripheral = peripheral => {

    // get the list of devices
    let {peripherals} = this.state;

    // if the device doesn't contain our board
    if (!peripherals.has(RFDuinoID)) {

      // then add it to the list of devices
      peripherals.set(RFDuinoID, peripheral);

      // and push it back to state
      this.setState({ peripherals })
    }
  };

  /**
   * This method will communicate to the board by passing a
   * series of commands
   * ==============================================================
   */
  write = () => {

    // 'b' => special command for starting data transferal
    const BCommand = stringToBytes('b');
    BleManager.write(
      RFDuinoID,
      '00002220-0000-1000-8000-00805F9B34FB',
      '00002222-0000-1000-8000-00805F9B34FB',
      BCommand
    ).catch(e => console.error("Error on write(): ", e));
  };

  /**
   * Return the entire component when it is mounted and ready
   * ==============================================================
   * @return {XML}
   */
  render() {
    return (
      <TouchableOpacity onPress={() => this.write()}>
        <Text>CLick me</Text>
      </TouchableOpacity>
    );
  }
}
