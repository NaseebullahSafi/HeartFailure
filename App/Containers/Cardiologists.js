import React, {Component} from 'react';
import {Text, View, TouchableOpacity, StyleSheet, TouchableHighlight, Image, Dimensions, ScrollView} from "react-native";
import Ionicons from "react-native-vector-icons/Feather";
import Svg, { Line, G, Path } from 'react-native-svg';
import _ from 'lodash';
import firebase from 'react-native-firebase';
import Database from '../Components/Database'
import User from '../Components/User';

export default class Cardiologists extends Component {
  constructor(props) {
    super(props);
    this.state = {
      User    : null,
      doctors : null
    };

    this.usersRef = firebase.app().database().ref('/Users');
    this.fetchDoctors = this.fetchDoctors.bind(this);
  }

  componentDidMount() {
    this.fetchDoctors(this.usersRef);
  }

  fetchDoctors = (userRef) => {
    User().then(user => {
      userRef.on('value', (snap) => {
        if (snap.val()) this.setState({
          type        : snap.val()[user.uid].type,
          name        : snap.val()[user.uid].name,
          healthAlert : _.has(snap.val()[user.uid], "healthAlrt") ? snap.val()[user.uid].healthAlert : "",
          doctors : JSON.parse(JSON.stringify(snap.val()), (k, v) => !v.type || (
            v.type === 'Doctor' && (_.has(snap.val()[user.uid], "Doctors") ? !_.has(snap.val()[user.uid].Doctors, k) : true )
          ) ? v : void 0)
        }, () => {})
      })
    });
  };

  doctors = () => {
    return this.state.doctors ? (
      Object.keys(this.state.doctors).map((uid, i) => {
        return (
          <View style={styles.doctorContainer} key={uid}>
            <View style={styles.imgCircleContainer}>
              <View>
                <Image style={styles.userImg} source={require('../Images/Naseebullah.jpg')} resizeMode="contain"/>
                <View style={styles.imgOverlay} />
              </View>
            </View>

            <View>
              <Text style={{textAlign: 'center', fontSize: 15, color: '#bccad0'}}>{this.state.doctors[uid].name}</Text>
              <Text style={{textAlign: 'center', fontSize: 12, color: 'rgba(188, 202, 208, 0.7)'}}>{this.state.doctors[uid].profession}</Text>
            </View>

            <TouchableOpacity style={styles.folllowBtn} onPress={() => Database.followDoctor(
              this.state.doctors[uid].name, this.state.doctors[uid].profession, uid,
              this.state.type, this.state.healthAlert
            )}>
              <Text style={{color: 'white', fontSize: 13, fontWeight: 'bold', textAlign: 'center'}}>Follow</Text>
            </TouchableOpacity>

          </View>
        )
      })
    ): null;
  };

  render() {
    return (
      <View style={styles.mainContainer}>
        <View style={styles.topSection}>
          <View style={styles.navigationBar}>
            <TouchableHighlight onPress={() => this.props.navigation.navigate('DrawerOpen')} activeOpacity={1.0} underlayColor="rgba(253,138,94,0)">
              <Svg height="24" width="24">
                <Line fill="none" stroke="#dfe0e1" strokeWidth="2" strokeLinecap="round" strokeLinejoin="round" x1="3" y1="12" x2="21" y2="12"/>
                <Line fill="none" stroke="#dfe0e1" strokeWidth="2" strokeLinecap="round" strokeLinejoin="round" x1="10.208" y1="6" x2="21" y2="6"/>
                <Line fill="none" stroke="#dfe0e1" strokeWidth="2" strokeLinecap="round" strokeLinejoin="round" x1="3" y1="18" x2="13.791" y2="18"/>
              </Svg>
            </TouchableHighlight>
          </View>
        </View>
        <ScrollView showsVerticalScrollIndicator={false}>
          <View style={{flexDirection: 'row', flexWrap: 'wrap', justifyContent: 'center', alignItems: 'center', backgroundColor: '#f7f8f9'}}>

            {this.doctors()}

          </View>
        </ScrollView>
      </View>
    )
  }
}


const styles = StyleSheet.create({
  mainContainer: {
    flex: 1,
    flexDirection: 'column',
    backgroundColor: '#f7f8f9',
    justifyContent: 'flex-start',
  },
  topSection: {
    flexDirection: 'column',
    backgroundColor: 'white',
    padding:10,
    position: 'relative',
    elevation: 0.2
  },
  navigationBar: {
    flexDirection: 'row',
    alignItems: 'stretch',
    justifyContent: 'space-between',
    padding: 10,
  },
  doctorContainer: {
    padding: 10,
    paddingTop: 40,
    paddingBottom: 20,
    width: 300,
    backgroundColor: 'white',
    margin: 10,
    borderRadius: 8,
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center'
  },
  imgCircleContainer: {
    position: 'relative',
    borderRadius: 80,
    height: 70,
    width: 70,
    borderColor: 'white',
    backgroundColor: 'white',
    alignItems: 'center',
    overflow: 'hidden',
    elevation: 10,
    justifyContent: 'center',
    marginBottom: 10
  },
  userImg: {
    borderRadius: 300,
    height: 70,
    width: 70,
  },
  imgOverlay: {
    position: 'absolute',
    backgroundColor: 'rgba(247, 248, 249, 0.7)',
    opacity: 0.5,
    borderRadius: 300,
    height: 70,
    width: 70,
  },
  folllowBtn: {
    marginTop: 10,
    backgroundColor: '#2fcc71',
    width: 100,
    padding: 5,
    borderRadius: 5,
    elevation: 3,
  }
});
