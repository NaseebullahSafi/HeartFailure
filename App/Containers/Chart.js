import React, {Component} from 'react'

// predefined component of react
import {Dimensions} from 'react-native'

// strict types for props
import PropTypes from 'prop-types'

// we need the highcharts support in react native
import ChartView from 'react-native-highcharts'

// this is important, the name should be specified as below
const Highcharts = "Highcharts"

/**
 * This component will take the config object from the props, and
 * pass it to highcharts to visualise. The object must of the
 * same structure as highcharts predefined obj
 * ==============================================================
 */
export default class Chart extends Component {
  constructor(props) {
    super(props);

    /**
     * Let us identify unique props for this component
     * @type {{
     *  type        : string, height  : Number, config: Object,
     *  chartLoading: boolean, options: {global: {useUTC: boolean}}
     * }}
     */
    this.state = {
      // the default type for the graph is day
      type: "day",

      // the chart needs to have specific height
      height: this.props.height,

      // this is the object that comes from other components
      config: this.props.config,

      // important for checking if the graph has loaded
      chartLoading: true,

      // highcharts default params for options
      options: {
        global: {
          useUTC: false
        }
      }
    }
  };

  /**
   * When the component loads, tell others that the component is
   * ready
   * ==============================================================
   */
  componentDidMount() {
    this.setState({loading: true});
  }

  /**
   * Default params for highcharts
   * ==============================================================
   */
  options = () => {
    this.setState({
      options: {
        global: {
          useUTC: false
        }
      }
    })
  };


  /**
   * When the component is ready, pass it to highcharts for
   * visualisation
   * ==============================================================
   * @return {XML}
   */
  render() {
    return(
      <ChartView
        style={{
          height: this.props.height,
          width: this.props.width || Dimensions.get('window').width
        }}
        config={this.props.config}
        options={this.props.options}
        // important if we want the chart to have scrollbar
        stock={this.props.stock || false}
        scalesPageToFit={false}
        automaticallyAdjustContentInsets={false}
        showsHorizontalScrollIndicator={false}
        horizontal={false}
        javaScriptEnabled={true}
        domStorageEnabled={true}
      />
    )
  }
}

// let us define some types for our props
Chart.propTypes = {
  height        : PropTypes.number.isRequired,
  config        : PropTypes.object.isRequired,
};
