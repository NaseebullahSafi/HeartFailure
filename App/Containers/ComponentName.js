import React, {Component} from 'react';

export default class ComponentName extends Component {
  constructor(props) {
    super(props);
    this.state = {...};
  }

  // called before the component is mounted
  componentWillMount() {};

  // called when the component is mounted
  componentDidMount = () => {};

  // called just before the component gets unmounted
  componentWillUnmount = () => {};

  // called when new props come
  componentWillReceiveProps = nextProps => {};

  // checks the props, if any changes, the component is re-rendered
  shouldComponentUpdate = ( nextProps, nextState, nextContext ) => {};

  // called just before it is updated
  componentWillUpdate = ( nextProps, nextState ) => {};

  // called when it is updated
  componentDidUpdate = (prevProps, prevState, prevContext) => {};

  // method that consists of the UI
  render() {
    return (null)
  }
};
