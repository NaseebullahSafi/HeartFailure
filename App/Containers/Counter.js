// predefined components from react
import React, {Component} from 'react'

// predefined components from react native
import {TouchableOpacity, Text, View} from 'react-native'

// feather icon package
import Feather from "react-native-vector-icons/Feather"

/**
 * This component is a button that is used for starting HS
 * recording
 * ==============================================================
 */
export default class Counter extends Component {
  constructor(props) {
    super(props);
    this.state = { counter: 0 };
    this.startRecording = this.startRecording.bind(this);
  }

  /**
   * This is a timer of the duration of the recording
   * ==============================================================
   * @param seconds
   */
  startRecording = seconds => {
    let count = (() => {});
    (count = (cc) => {
      this.setState({counter: cc});
      if (cc > 0) setTimeout(function() { count(--cc); }, 1000);
    })(seconds);
  };

  /**
   * When the component is ready, show the button
   * ==============================================================
   * @return {XML}
   */
  render() {
    return (
      <View style={{alignItems: 'flex-end'}}>
        <TouchableOpacity
          onPress={() => this.startRecording(30)}
          style={{
            width: 50,
            height: 50,
            borderRadius: 300,
            backgroundColor: '#E67D8F',
            alignItems: 'center',
            justifyContent: 'center',
            elevation: 5, marginRight: 30
          }}>
          {
            this.state.counter === 0 ? (
              <Feather
                name={"mic"}
                size={20}
                color={"white"}
              />
            ): <Text
              style={{
                fontWeight: 'bold',
                color: 'white',
                fontSize: 17
              }}>{this.state.counter}</Text>
          }
        </TouchableOpacity>
      </View>
    )
  }
}
