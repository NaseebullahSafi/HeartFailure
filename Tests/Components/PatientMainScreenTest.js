// need to import React itself
import React from 'react'

// everything fro react native
import 'react-native'

// required for mocking modules
import Enzyme,{ shallow } from 'enzyme'

// needed to pass components to the renderer
import renderer from 'react-test-renderer'

// adapter is needed for this react version
import Adapter from 'enzyme-adapter-react-16';

// we cant use firebase in jest, so we have mocked it
import firebase from '../../firebase'

// the component that will be tested
import {PatientMainScreen} from "../../App/Containers/PatientMainScreen";

Enzyme.configure({ adapter: new Adapter() });

/**
 * Test the entire component to see if it renders properly
 */
describe('Testing <PatientMainScreen /> component', () => {

  // create an image of the component, and match it with the snapshot
  it('renders as expected', () => {
    const tree = renderer.create(
      <PatientMainScreen />
    ).toJSON();
    expect(tree).toMatchSnapshot()
  });

  /**
   * Set of methods from the component is tested below
   */
  it('should request User API from firebase', () => {
    const wrapper = shallow(<PatientMainScreen />);
    // attach the user reference from firebase to the component
    let userRef = firebase.database().ref(`/Users/`);

    // attach the ECG reference from firebase to the component
    let ecgRef = firebase.database().ref(`/ECG/`);

    // attach the health reference from firebase to the component
    let healthRef = firebase.database().ref(`/Health/`);

    const inst = wrapper.instance().fetchData(
      userRef, ecgRef, healthRef, {}
    );
    expect(inst)
  });

  it('should apply letter spacing', () => {
    const wrapper = shallow(<PatientMainScreen />);
    const inst = wrapper.instance().applyLetterSpacing("Doctor");
    expect(inst).toBe("Doctor".split('').join('\u200A'.repeat(1)))
  });

  it('should sign out', () => {
    const wrapper = shallow(<PatientMainScreen />);
    const inst = wrapper.instance().signOutUser();
    expect(inst)
  });

  it('should toggle buttons', () => {
    const wrapper = shallow(<PatientMainScreen />);
    const inst = wrapper.instance().toggleBtns("EditView", true);
    expect(inst)
  });

  it('should create config object for highcharts', () => {
    const wrapper = shallow(<PatientMainScreen />);
    const inst = wrapper.instance().config();
    expect(inst)
  });

  it('should create temperature object for highcharts', () => {
    const wrapper = shallow(<PatientMainScreen />);
    const inst = wrapper.instance().temperature;
    expect(inst)
  });

  it('should create calories object for highcharts', () => {
    const wrapper = shallow(<PatientMainScreen />);
    const inst = wrapper.instance().caloriesBurned;
    expect(inst)
  });

  it('should create fat object for highcharts', () => {
    const wrapper = shallow(<PatientMainScreen />);
    const inst = wrapper.instance().fatBurned;
    expect(inst)
  });

  it('should create heart rate object for highcharts', () => {
    const wrapper = shallow(<PatientMainScreen />);
    const inst = wrapper.instance().heartRate;
    expect(inst)
  });

  it('should render input fields for the form', () => {
    const wrapper = shallow(<PatientMainScreen />);
    const inst = wrapper.instance().renderInput({input: {...jest.fn()}});
    expect(inst)
  });

  it('should open the side menu for messages', () => {
    const navigation = { navigate: jest.fn() };
    const wrapper = shallow(<PatientMainScreen navigation={navigation}/>);
    const inst = wrapper.instance().openSideMenu(navigation.navigate);
    expect(inst)
  });

  it('should render the top container', () => {
    const wrapper = shallow(<PatientMainScreen />);
    const inst = wrapper.instance().topContainer({});
    expect(inst)
  });

  it('should render the about container when in default view', () => {
    const wrapper = shallow(<PatientMainScreen />);
    const inst = wrapper.instance().aboutDefaultView({});
    expect(inst)
  });

  it('should render the about container when in edit view', () => {
    const wrapper = shallow(<PatientMainScreen handleSubmit={jest.fn()} />);
    const inst = wrapper.instance().aboutEditView({});
    expect(inst)
  });

  it('should create a container for about', () => {
    const wrapper = shallow(<PatientMainScreen />);
    const inst = wrapper.instance().about({});
    expect(inst)
  });

  it('should render ECG data', () => {
    const wrapper = shallow(<PatientMainScreen />);
    const inst = wrapper.instance().$ECG({}, [2,1,2], "Patient");
    expect(inst)
  });

});

