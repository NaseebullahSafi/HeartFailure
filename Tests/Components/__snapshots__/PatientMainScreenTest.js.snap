// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <PatientMainScreen /> component renders as expected 1`] = `
<RCTScrollView
  showsVerticalScrollIndicator={false}
>
  <View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "column",
          "justifyContent": "flex-start",
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "alignSelf": "center",
              "flexDirection": "column",
              "paddingTop": 20,
            },
            Object {
              "width": "95%",
            },
          ]
        }
      />
    </View>
  </View>
</RCTScrollView>
`;
